manifest_test.go

```diff
@@ -259,3 +302,117 @@ func randomBytes(p []byte) {
 		p[i] = byte(rand.Intn(1<<8 - 1))
 	}
 }
+
+// expectedResourceList sorts the set of resources into the order
+// expected in the manifest and collapses hardlinks
+func expectedResourceList(resources []dresource) ([]Resource, error) {
+	resourceMap := map[string]Resource{}
+	paths := []string{}
+	for _, r := range resources {
+		absPath := r.path
+		if !filepath.IsAbs(absPath) {
+			absPath = "/" + absPath
+		}
+		uidStr := strconv.Itoa(r.uid)
+		gidStr := strconv.Itoa(r.uid)
+		switch r.kind {
+		case rfile:
```

Comment:
So, it would be nice to get rid of these test types. I actually made them before the advent of the default package schema. I think this use case could be filled in with a declarative resource struct, coupled with a method that will select the correct wrapping type:

```go
r, err := Declare(Resource{
    Kind: RegularFile,
    Path: "/foo",
})
```

And for inline declarations for making tests:

```go
[]Resource{
    MustDeclare(Resource{
        Kind: RegularFile,
        Path: "/foo",
    }),
}
```